paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PaparazziPlugin.kt

```diff
@@ -157,17 +157,23 @@ public class PaparazziPlugin : Plugin<Project> {
         task.compileSdkVersion.set(android.compileSdkVersion())
         task.projectResourceDirs.set(
           project.provider {
-            val resourcesComputer = variant.mergeResourcesProvider?.get()?.resourcesComputer
-            val generateResValuesDirs = resourcesComputer?.generatedResOutputDir ?: project.files()
-            val extraGeneratedResDirs = resourcesComputer?.extraGeneratedResFolders ?: project.files()
-
-            extraGeneratedResDirs.map(projectDirectory::relativize) + localResourceDirs.relativize(projectDirectory) + generateResValuesDirs.map(projectDirectory::relativize)
-          }
+            val resourcesComputer = variant.mergeResourcesProvider
+            val extraGeneratedResDirs =
+              resourcesComputer?.map { it.resourcesComputer.extraGeneratedResFolders }
+                ?: project.provider { project.files() }
+            val generatedResOutputDirs =
+              resourcesComputer?.map { it.resourcesComputer.generatedResOutputDir }
+                ?: project.provider { project.files() }
+            extraGeneratedResDirs
+              .zip(project.provider { localResourceDirs }) { t, u -> t.plus(u) }
+              .zip(generatedResOutputDirs) { t, u -> t.plus(u) }
+              .flatMap { it.relativize(projectDirectory) }
```

Comment:
this might read better (if compiles)
```suggestion
            extraGeneratedResDirs
              .zip(project.provider { localResourceDirs }, FileCollection::plus)
              .zip(generatedResOutputDirs, FileCollection::plus)
              .flatMap { it.relativize(projectDirectory) }
```